src/main/java/fm/last/moji/impl/FileLengthCommand.java

```diff
@@ -102,6 +102,12 @@ private long getContentLength(HttpURLConnection httpConnection) {
     long length = 0;
     String rawLength = httpConnection.getHeaderField("Content-Length");
 
+    // Some webdav server, e.g., apache, does not return content-length header when the file size is zero
+    if(rawLength == null) {
```

Comment:
1. Why zero length files
If you want to retain file tree structure in mogilefs, you may use zero length files to stand for directories, e.g., /var/www/.

2. content-length handling
According to HTTP spec., content-length header is not necessary. There are some discussions here: http://stackoverflow.com/questions/3854842/content-length-header-with-head-requests

Indicate no content-length response to zero length file is reasonable for legal http/webdav server I think.

Another work around is that to implement an check file size command in mogilefsd, so we can get the file size from mogilefs DB.